be/src/exprs/vectorized/json_functions.cpp

```diff
@@ -322,43 +322,63 @@ ColumnPtr JsonFunctions::_iterate_rows(FunctionContext* context, const Columns&
     return result.build(ColumnHelper::is_all_const(columns));
 } // namespace starrocks::vectorized
 
-template <PrimitiveType primitive_type>
-void JsonFunctions::_build_column(ColumnBuilder<primitive_type>& result, simdjson::ondemand::value& value) {
-    if constexpr (primitive_type == TYPE_INT) {
-        int64_t i64;
-        auto err = value.get_int64().get(i64);
-        if (UNLIKELY(err)) {
-            result.append_null();
-            return;
-        }
+template <>
+void JsonFunctions::_build_column(ColumnBuilder<TYPE_INT>& result, simdjson::ondemand::value& value) {
+    int64_t i64;
+    auto err = value.get_int64().get(i64);
+    if (UNLIKELY(err)) {
+        result.append_null();
+        return;
+    }
+
+    result.append(i64);
+    return;
+}
 
-        result.append(i64);
+template <>
+void JsonFunctions::_build_column(ColumnBuilder<TYPE_DOUBLE>& result, simdjson::ondemand::value& value) {
+    double d;
+    auto err = value.get_double().get(d);
+    if (UNLIKELY(err)) {
+        result.append_null();
         return;
+    }
 
-    } else if constexpr (primitive_type == TYPE_DOUBLE) {
-        double d;
-        auto err = value.get_double().get(d);
-        if (UNLIKELY(err)) {
-            result.append_null();
-            return;
-        }
+    result.append(d);
+    return;
+}
 
-        result.append(d);
+template <>
+void JsonFunctions::_build_column(ColumnBuilder<TYPE_VARCHAR>& result, simdjson::ondemand::value& value) {
+    simdjson::ondemand::json_type tp;
+    auto err = value.type().get(tp);
+    if (UNLIKELY(err)) {
```

Comment:
How about simplify it with a macro?

Reply:
Great idea. I have replaced it with a macro `APPEND_NULL_AND_RETURN_IF_ERROR`